.github/workflows/_docker-build-push.yml

```diff
@@ -48,14 +48,28 @@ jobs:
         id: regex
         with:
           text: ${{ github.ref }}
-          regex: '^(refs\/tags\/${{ inputs.service-name }}\/v([^\/]+))|(refs\/heads\/(main))$'
+          # https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+          regex: '^(refs\/tags\/${{ inputs.service-name }}\/v((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?))|(refs\/heads\/(main))$'
 
       - name: Extract tag name
         id: tags_extractor
         run: |
-          t=${{ steps.regex.outputs.group2 }}
-          m=${{ steps.regex.outputs.group4 }}
-          (if ! [[ "$t" == "" ]]; then echo tags=${{ env.IMAGE_NAME }}:$t, ${{ env.IMAGE_NAME }}:latest; elif ! [[ "$m" == "" ]]; then echo tags=${{ env.IMAGE_NAME }}:$m; else echo tags=; fi) >> $GITHUB_OUTPUT
+          version=${{ steps.regex.outputs.group2 }}
```

Comment:
probably capture groups also need to be adjusted? now major, minor, and patch have their own separate groups 👀

Reply:
not sure how it works though

Reply:
> (so cg1 = major, cg2 = minor, cg3 = patch, cg4 = prerelease and cg5 = buildmetadata)

Reply:
u can check how groups are located here:

https://regex101.com/r/DgoEdc/1

`group2` is whole tag with major, minor, patch, prerelease, buildmetadata

Reply:
ah, I mixed up `()` and `[]` 😅 
ok, nice!